### PR TITLE
Updated the docs for Error::description

### PR DESCRIPTION
**Pull request by omern1**
Fixes https://github.com/rust-lang/rust/issues/37163

### CODE REVIEW

File: src/libstd/error.rs [hunk 1]
```diff
@@ -65,9 +65,11 @@ use string;
...
     /// A short description of the error.
     ///
-    /// The description should not contain newlines or sentence-ending
-    /// punctuation, to facilitate embedding in larger user-facing
-    /// strings.
+    /// The description should only be used for a simple message.
+    /// It should not contain newlines or sentence-ending punctuation,
+    /// to facilitate embedding in larger user-facing strings.
+    /// For showing formatted error messages with more information see
+    /// (Display)[https://doc.rust-lang.org/std/fmt/trait.Display.html].
```
Comment (nabeelomer) [inline src/libstd/error.rs]: Thanks

File: src/libstd/error.rs [hunk 2]
```diff
@@ -65,9 +65,11 @@ use string;
...
     /// A short description of the error.
     ///
-    /// The description should not contain newlines or sentence-ending
-    /// punctuation, to facilitate embedding in larger user-facing
-    /// strings.
+    /// The description should only be used for a simple message.
+    /// It should not contain newlines or sentence-ending punctuation,
+    /// to facilitate embedding in larger user-facing strings.
+    /// For showing formatted error messages with more information see
+    /// [Display](https://doc.rust-lang.org/std/fmt/trait.Display.html).s
```
Comment (frewsxcv) [inline src/libstd/error.rs]: Might want to remove the 's' at the end of this line.

Comment (nabeelomer) [inline src/libstd/error.rs]: Thanks.

**Comment by bluss** [comment]
Thank you!

@bors r+ rollup

**Comment by nabeelomer** [comment]
@bluss This looks like a problem with the buildbot's git repository
```
fatal: Unable to create '/Users/rustbuild/src/rust-buildbot/slave/auto-mac-64-opt-rustbuild/build/.git/modules/src/llvm/index.lock': File exists.

If no other git process is currently running, this probably means a
git process crashed in this repository earlier. Make sure no other git
process is running and remove the file manually to continue.
```

**Comment by bluss** [comment]
Yes, it just happened to a whole bunch of PRs unfortunately. We'll retry this as soon as it's working again.

**Pull request merged**